### DISCUSSION TITLE
List of commands for zerotier-cli

### DISCUSSION THREAD

**Discussion by Urajmal** [sinamics/ztnet#241] [Q&A]
Hi again! 

While you can list peers and list and join networks, I have nowhere on the Internet found, how can I approve peers, list routes, add routes, add DNS, change subnet network. Is that something that is not possible with zerotier-cli, or is that written and controlled somewhere else?

With different question; if I have only CLI access, can I control those things with some commands? 

Thank you!

**Comment by sinamics** [comment]
I dont think the cli supports all these commands, so you would need to use the API interface. 

Get the `AUTH_SECRET` from `/var/lib/zerotier-one/authtoken.secret` 
Example, something like this should set a member to Authorized: ( Not tested )
```bash
curl -X POST \
     -H 'Content-Type: application/json' \
     -H 'X-ZT1-Auth {AUTH_SECRET}' \
     -d '{"config":{"authorized":true}}' \
     http://{YOUR_LOCAL_CONTROLLER_URL}/controller/network/{YOUR_NETWORK_ID}/member/{MEMBER_ID}
```

You can find most of the available endpoints here, but its not a complete list.
https://docs.zerotier.com/service/v1/
